selendroid-standalone/src/main/java/io/selendroid/standalone/server/SelendroidStandaloneServer.java

```diff
@@ -68,11 +66,41 @@ protected SelendroidStandaloneDriver initializeSelendroidServer() throws Android
   public void start() {
     webServer.start();
     if (config.isGrid()) {
-      try {
-        new SelfRegisteringRemote(config, driver).performRegistration();
-      } catch (Exception e) {
-        log.log(Level.SEVERE, "Error registering selendroid into grid hub.", e);
+
+      class GridRegistrationTask extends TimerTask {
+        private final SelendroidStandaloneDriver driver;
+        private final SelendroidConfiguration configuration;
+        private final Logger log = Logger.getLogger(GridRegistrationTask.class.getName());
+        private boolean firstTime = true;
+
+        GridRegistrationTask(SelendroidStandaloneDriver driver, SelendroidConfiguration configuration) {
+          this.driver = driver;                // passing reference to driver and config
+          this.configuration = configuration;  // as, let's say, config might change during runtime... (device quantity)
+        }
+
+        @Override
+        public void run() {
+          try {
+            final SelfRegisteringRemote remote = new SelfRegisteringRemote(configuration, driver);
+
+            if (firstTime) {
```

Comment:
@asm89, the check for firstTime is a kind of '--force-register' mode. The filed use for this is - let's say, you would like to restart selendroid for some reason (upgrade or config changes - emulators/devices/other grid parameters). And selendroid restarted quite fast, so grid hub have not noticed that selendroid node was absent. Then without this '--force-register' mode, the selendroid node will not send new config to grid hub.